addons/hr_holidays/models/hr_leave_type.py

```diff
@@ -111,72 +102,20 @@ def _search_valid(self, operator, value):
                 ('validity_stop', signs[0] if value else signs[1], dt),
                 ('validity_start', signs[1] if value else signs[0], dt)]
 
-    @api.multi
-    def get_days(self, employee_id):
-        # need to use `dict` constructor to create a dict per id
-        result = dict((id, dict(max_leaves=0, leaves_taken=0, remaining_leaves=0, virtual_remaining_leaves=0)) for id in self.ids)
-
-        requests = self.env['hr.leave'].search([
-            ('employee_id', '=', employee_id),
-            ('state', 'in', ['confirm', 'validate1', 'validate']),
-            ('holiday_status_id', 'in', self.ids)
-        ])
-
-        allocations = self.env['hr.leave.allocation'].search([
-            ('employee_id', '=', employee_id),
-            ('state', 'in', ['confirm', 'validate1', 'validate']),
-            ('holiday_status_id', 'in', self.ids)
-        ])
-
-        for request in requests:
-            status_dict = result[request.holiday_status_id.id]
-            status_dict['virtual_remaining_leaves'] -= request.number_of_days_temp
-            if request.state == 'validate':
-                status_dict['leaves_taken'] += request.number_of_days_temp
-                status_dict['remaining_leaves'] -= request.number_of_days_temp
-
-        for allocation in allocations:
-            status_dict = result[allocation.holiday_status_id.id]
-            if allocation.state == 'validate':
-                # note: add only validated allocation even for the virtual
-                # count; otherwise pending then refused allocation allow
-                # the employee to create more leaves than possible
-                status_dict['virtual_remaining_leaves'] += allocation.number_of_days_temp
-                status_dict['max_leaves'] += allocation.number_of_days_temp
-                status_dict['remaining_leaves'] += allocation.number_of_days_temp
-
-        return result
-
-    @api.multi
-    def _compute_leaves(self):
-        data_days = {}
-        if 'employee_id' in self._context:
-            employee_id = self._context['employee_id']
-        else:
-            employee_id = self.env['hr.employee'].search([('user_id', '=', self.env.user.id)], limit=1).id
-
-        if employee_id:
-            data_days = self.get_days(employee_id)
-
-        for holiday_status in self:
-            result = data_days.get(holiday_status.id, {})
-            holiday_status.max_leaves = result.get('max_leaves', 0)
-            holiday_status.leaves_taken = result.get('leaves_taken', 0)
-            holiday_status.remaining_leaves = result.get('remaining_leaves', 0)
-            holiday_status.virtual_remaining_leaves = result.get('virtual_remaining_leaves', 0)
-
     @api.multi
     def name_get(self):
-        if not self._context.get('employee_id'):
+        employee_id = self.env['hr.employee'].browse(self._context.get('employee_id', 1))
```

Comment:
Not sure to understand why you set a fallback to `1` in
`self._context.get('employee_id', 1)`

This mean `employee_id ` will always be filled if there is an employee with as ID 1 (which is not guaranteed by the way, user with id 1 is guaranteed, but not employee).

If not `employee_id` is not in the context, you should just keep the previous behavior, to have the regular name_get without the remaining leaves rather than displaying the remaining leave of another employee :dagger: 

Besides I would do the `browse` later, to avoid doing it for no reason and waste some resources as it could not be used if `employee_id` is not set in the context

Reply:
My bad for the fallback, I was testing something and forgot to remove it.